### DISCUSSION TITLE
What's missing in the contributing documentation?

### DISCUSSION THREAD

**Discussion by konstin** [astral-sh/ruff#5693] [General]
Are there things that are missing in the contributing documentation, parts that you need help understanding, explanations that are unclear? Report them here!

**Comment by charliermarsh** [comment]
Maybe better as a Discussion so that it can be long-lived?

**Reply by konstin** [reply]
good idea, didn't think about that

**Comment by dhruvmanila** [comment]
A new example section explaining on how to go about making a rule auto-fixable. This would include, but not limited to, the following:
* Using the `checker.patch(...)` method to check if the user wants to generate the patch which involves importing the `AsRule` trait:
	
	```rust
	use crate::registry::AsRule;
	
	pub fn check_rule() {
		// ...
		let mut diagnotic = Diagnostic::new(...)
		if checker.patch(diagnostic.rule.kind()) {
			// ...
		}
		checker.diagnostics.push(diagnostic)
	}
* Types of `Edit` (`insertion`, `deletion`, `replacement` and their ranged variant)
* Setting the `Applicability`
* Generating the edits using:
	* `libcst`
	* Using the generator API (`checker.generator().expr(...)`, `checker.generator().stmt(...)`)
	* Manually generating the source string
* ...
